core/server/data/migrations/versions/4.4/02-migrate-members-signup-access.js

```diff
@@ -0,0 +1,126 @@
+const logging = require('../../../../../shared/logging');
+const {createTransactionalMigration} = require('../../utils');
+const MembersConfigProvider = require('../../../../services/members/config');
+const settingsCache = require('../../../../services/settings/cache');
+const config = require('../../../../../shared/config');
+
+// members_allow_free_signup was migrated to members_signup_access in 4.3/08-migrate-members-signup-setting
+// using a direct binary migration of true=all and false=invite based on new setting behaviour matching the
+// old setting. That was innacurate and the behaviour is changing to a proper invite-only toggle which
+// requires a more nuanced migration.
+//
+// 'all' - no need to change, matches expected behaviour
+// 'invite' - change to 'all' based on members usage
+// 'none' - no need to change, it's a completely new option that wasn't previously migrated to
+//
+// For the majority of cases 'invite' should actually be 'all'. The true invite-only case is:
+// - stripe not configured
+// - portal configured with no plans
+//
+// When switching from signup access from 'invite' to 'all' we also need to remove the 'free' plan from
+// portal settings to avoid paid-only sites suddenly offering free plan signups
+
+module.exports = createTransactionalMigration(
+    async function up(connection) {
+        const currentSetting = await connection('settings')
+            .where('key', 'members_signup_access')
+            .select('value')
+            .first();
+
+        if (!currentSetting) {
+            logging.info('Skipping update of members_signup_access setting. Does not exist');
+            return;
+        }
+
+        if (currentSetting.value !== 'invite') {
+            logging.info(`Skipping update of members_signup_access setting. Not set to 'invite', keeping current value`);
+            return;
+        }
+
+        const membersConfig = new MembersConfigProvider({
+            settingsCache,
+            config
+        });
+
+        const hasStripe = membersConfig.isStripeConnected();
```

Comment:
🤔 Just wondering if we should look into extracting this logic out directly via settings/config in the migration than depend on `membersConfig`, my only concern being any future changes to membersConfig or this method might break the migration. Ideally any updates to members config would mean this piece of code is updated as well so maybe its fine to leave as-is ?

Reply:
Yeah, I had similar concerns but it seemed like checking if Stripe is configured touched both settings/config with some additional logic which also seemed brittle to have in a migration.

Extracting the check could be useful but would that gain much? If `membersConfig` changes I'd expect usage of it to be checked too 🤔

Reply:
Yeah that's fair, and its more likely we continue to have a method to check if stripe is connected so any changes can update this on usage check